src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -10,7 +10,57 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
+        internal static unsafe string getcwd()
+        {
+            const int StackLimit = 256;
+
+            // First try to get the path into a buffer on the stack
+            byte* stackBuf = stackalloc byte[StackLimit];
+            string result = getcwd(stackBuf, StackLimit);
+            if (result != null)
+                return result;
+
+            // If that was too small, try increasing large buffer sizes
+            // until we get one that works or until we hit MaxPath.
+            int maxPath = Interop.libc.MaxPath;
+            int bufferSize = StackLimit;
+            while ((bufferSize = Math.Min(bufferSize * 2, maxPath)) < maxPath)
```

Comment:
This looks to me like it never tries with maxPath.

Assuming maxPath is 4096 the loop will do { 512 => try, 1024 => try, 2048 => try, 4096 => abort, not < maxPath }.

If someone had rounded their system off to 5000 characters, you'd try the 4096, then the next iteration should try 5000, but won't.

```C#
do
{
    bufferSize = Math.Min(bufferSize * 2, maxPath);

    ...
} while (bufferSize < maxPath)
```

perhaps?  It does one extra call in the degenerate case where maxPath <= StackLimit, which could be guarded against if it's really a concern.